src/flb_storage.c

```diff
@@ -241,7 +266,7 @@ static void cb_storage_metrics_collect(struct flb_config *ctx, void *data)
     msgpack_packer_init(&mp_pck, &mp_sbuf, msgpack_sbuffer_write);
 
     /* Pack main map and append relevant data */
-    msgpack_pack_map(&mp_pck, 2);
+    msgpack_pack_map(&mp_pck, 3);
```

Comment:
It is hard to know why map size is 3.

It may be better `flb_mp_*` API that can change map size after creating key/vals.
Could you check below test code to understand how to use `flb_mp_*` API ?
https://github.com/fluent/fluent-bit/blob/v1.9.4/tests/internal/mp.c#L40